src/database/entities/BaseEntity.ts

```diff
@@ -0,0 +1,13 @@
+import { PrimaryKey, Property } from "@mikro-orm/core";
+import { v4 as uuidv4 } from "uuid";
+
+export abstract class BaseEntity {
+  @PrimaryKey()
+  id: number | string = uuidv4();
```

Comment:
per discord documentation, userId will always be string. So no need to include number here

Reply:
this is a base entity which will be used for more than just users. a union type here prevents the TS compiler from freaking out if we decide to user numerical auto incrementing ids on another entity